test/whitequark/test_context_lambda.rb

```diff
@@ -0,0 +1,4 @@
+# typed: strict
+
+module A; ->(a = (return; 1)) {}; end
+class B; ->(a = (return; 1)) {}; end
```

Comment:
Did you confirm that this test did something different / incorrect before this change?

Reply:
The link given in the introduction shows the behavior with current Sorbet: https://sorbet.run/#%23%20typed%3A%20true%0A%0Aclass%20A%3B%20-%3E(a%20%3D%20(return%3B%201))%20%7B%7D%3B%20end